preload.js

```diff
@@ -199,8 +199,12 @@ const SharedownAPI = (() => {
 
             page.setDefaultNavigationTimeout(puppyTimeout);
 
-            await page.goto(video.url, {waitUntil: 'networkidle0'});
-            await _sharepointLogin(page, loginData);
+            if(_loginModule.shouldBypass()){
```

Comment:
i would prefer a cleaner solution

for ex.
no need to add an _if_ here, just keep the original code

add a new function in Basic module `getLoginModuleID()` to return the name of the class
your login module would be an empty class
make the function visible to the app in https://github.com/kylon/Sharedown/blob/300f0a4cdb3b8cac4ed0e63c34ca3b8c55699b6b/preload.js#L36

you can add a check here:
https://github.com/kylon/Sharedown/blob/300f0a4cdb3b8cac4ed0e63c34ca3b8c55699b6b/sharedown/utils.js#L41

at the beginning of the function
```
if (_sharedownApi.sharedownLoginModule.getLoginModuleID() === 'IDofyourModule')
        return {};
```

and the last check here:
https://github.com/kylon/Sharedown/blob/300f0a4cdb3b8cac4ed0e63c34ca3b8c55699b6b/preload.js#L69

at the beginning of the function
```
if (!Object.keys(logData).length)
      return;
```

i did not test the actual code but something like this should work